routes/delete-file.js

```diff
@@ -35,7 +35,7 @@ module.exports = function(config) {
         return;
       }
 
-      delete req.session.project.files[path];
+      req.session.project.files.splice(req.session.project.files.indexOf(existingFile.id), 2);
```

Comment:
Can you move the `splice` stuff into `utils` too?  Let's hide all this nonsense of packing the files from callers.